cpp/test/optionsTest.cc

```diff
@@ -0,0 +1,275 @@
+  newPolicy = openFileOptions->getCacheLocationPolicy();
+  policyClass = openFileOptions->getCacheLocationPolicyClass();
+  assert(policyClass.compare(
+      "alluxio.client.file.policy.MostAvailableFirstPolicy") == 0);
+  openFileOptions->setCacheLocationPolicyClass(
+      "alluxio.client.file.policy.RoundRobinPolicy");
+  policyClass = openFileOptions->getCacheLocationPolicyClass();
+  assert(policyClass.compare(
+      "alluxio.client.file.policy.RoundRobinPolicy") == 0);
+  openFileOptions->setReadType(new ReadType(2));
+  ReadType* readType = openFileOptions->getReadType();
+  assert(readType->getValue() == 2);
+  openFileOptions->setMaxUfsReadConcurrency(3);
+  int maxUfsReadConcurrency = openFileOptions->getMaxUfsReadConcurrency();
+  assert(maxUfsReadConcurrency == 3);
+  BlockLocationPolicy* bPolicy = NULL;
+  bPolicy = DeterministicHashPolicy::getPolicy();
+  openFileOptions->setUfsReadLocationPolicy(bPolicy);
+  BlockLocationPolicy* nPolicy = openFileOptions->getUfsReadLocationPolicy();
+  policyClass = openFileOptions->getUfsReadLocationPolicyClass();
+  assert(policyClass.compare(
+      "alluxio.client.block.policy.DeterministicHashPolicy") == 0);
+  openFileOptions->setUfsReadLocationPolicyClass(
+      "alluxio.client.file.policy.RoundRobinPolicy");
+  policyClass = openFileOptions->getUfsReadLocationPolicyClass();
+  assert(policyClass.compare(
+      "alluxio.client.file.policy.RoundRobinPolicy") == 0);
+}
+
+void SetAttributeOptionsTest() {
+  SetAttributeOptions* setAttributeOptions =
+      SetAttributeOptions::getDefaultOptions();
+  setAttributeOptions->setPinned(true);
+  bool pinned = setAttributeOptions->getPinned();
+  assert(pinned == true);
+  setAttributeOptions->setTtl(10);
+  int64_t ttl = setAttributeOptions->getTtl();
+  assert(ttl == 10);
+  setAttributeOptions->setTtlAction(new TtlAction("FREE"));
+  TtlAction* ttlAction = setAttributeOptions->getTtlAction();
+  assert(ttlAction->isFree() == true);
+  setAttributeOptions->setPersisted(true);
+  bool persisted = setAttributeOptions->getPersisted();
+  assert(persisted == true);
+  Mode* mode = Mode::getDefaultMode();
+  mode->setOwnerBits(new Bits("rwx"));
+  setAttributeOptions->setMode(mode);
+  Mode* newMode = setAttributeOptions->getMode();
+  assert(newMode->getOwnerBits()->toString().compare("rwx") == 0);
+  setAttributeOptions->setRecursive(true);
+  bool recursive = setAttributeOptions->isRecursive();
+  assert(recursive == true);
+  Status status1 = setAttributeOptions->setOwner("user1");
+  Status status2 = setAttributeOptions->setGroup("group1");
+  assert(status1.ok());
+  assert(status2.ok());
+  std::string owner = setAttributeOptions->getOwner();
+  std::string group = setAttributeOptions->getGroup();
+  assert(owner.compare("user1") == 0);
+  assert(group.compare("group1") == 0);
+}
+
+void GetStatusOptionsTest() {
+  GetStatusOptions* getStatusOptions = GetStatusOptions::getDefaultOptions();
```

Comment:
`getStatusOptions` has to be `delete`ed since it is allocated on heap by `new` in `getDefaultOptions`.

Reply:
I have tried to add some "delete" to fix this and it passed the local test, but it failed to pass tests when I pushed it to remote server. Even if I rollback to the version which has passed the remote tests before, it cannot pass the test now. Then I refer to the log, the error is as below:
Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:1.8:run (make) on project cpp: An Ant BuildException has occured: Execute failed: java.io.IOException: Cannot run program "cmake" (in directory "/usr/src/alluxio/cpp/target/native"): error=2, No such file or directory.
It seems the Alluxio remote server has not installed "cmake"?

Reply:
@uronce-cc 
The error log: Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:1.8:run (make) on project cpp: An Ant BuildException has occured: Execute failed: java.io.IOException: Cannot run program "cmake" (in directory "/usr/src/alluxio/cpp/target/native"): error=2, No such file or directory.
It seems the Alluxio remote server has not installed "cmake"?